js/data/abstract_store.js

```diff
@@ -92,6 +93,11 @@ const Store = Class.inherit({
     },
 
     _loadImpl: function(options) {
+        if(!isEmptyObject(this._compareOptions)) {
+            options = options || {};
+            options._compareOptions = { ...this._compareOptions, ...options._compareOptions };
+        }
```

Comment:
Nothing scary will happen if we spread the empty object :)

Let's simplify these lines:
```suggestion
        options._compareOptions = { ...this._compareOptions, ...options._compareOptions };
```

Reply:
if so we get two spread operations and reassign when it not necessary